.github/workflows/mac-pkg.yml

```diff
@@ -0,0 +1,111 @@
+name: Sign and Upload Mac Installer
+
+on:
+  release:
+    types: [created, published]
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Release version to build and upload (e.g. "v4.2.1")'
+        required: true
+jobs:
+  build:
+    runs-on: macos-latest
+    env:
+      APPLICATION_CERTIFICATE: ${{ secrets.MACOS_APPLICATION_CERT }}
+      CODESIGN_IDENTITY: ${{ secrets.MACOS_APPLICATION_IDENTITY }}
+      INSTALLER_CERTIFICATE: ${{ secrets.MACOS_INSTALLER_CERT }}
+      PRODUCTSIGN_IDENTITY: ${{ secrets.MACOS_INSTALLER_IDENTITY }}
+      CERTIFICATE_PWD: ${{ secrets.MACOS_CERTIFICATE_PWD }}
+
+      NOTARIZE_TEAM: ${{ secrets.MACOS_NOTARIZATION_TEAM_ID }}
+      NOTARIZE_USERNAME: ${{ secrets.MACOS_NOTARIZATION_APPLE_ID }}
+      NOTARIZE_PASSWORD: ${{ secrets.MACOS_NOTARIZATION_PWD }}
+
+      KEYCHAIN_PWD: ${{ secrets.MACOS_CI_KEYCHAIN_PWD }}
+    steps:
+    - name: Determine Version
+      id: getversion
+      run: |
+        if [ -z "${{ inputs.version }}" ]
+        then
+              VERSION=${{ github.event.release.tag_name }}
+        else
+              VERSION=${{ inputs.version }}
+        fi
+        echo
+        echo "version=$VERSION" >> $GITHUB_OUTPUT
+    - name: Check uploads
+      id: check
+      run: |
+        URI="https://github.com/containers/podman/releases/download/${{steps.getversion.outputs.version}}"
+        ARM_FILE="podman-installer-macos-arm64.pkg"
+        AMD_FILE="podman-installer-macos-amd64.pkg"
+
+        status=$(curl -s -o /dev/null -w "%{http_code}" "${URI}/${ARM_FILE}")
+        if [[ "$status" == "404" ]] ; then
+          echo "buildarm=true" >> $GITHUB_OUTPUT
+        else
+          echo "ARM installer already exists, skipping"
+          echo "buildarm=false" >> $GITHUB_OUTPUT
+        fi
+
+        status=$(curl -s -o /dev/null -w "%{http_code}" "${URI}/${AMD_FILE}")
+        if [[ "$status" == "404" ]] ; then
+          echo "buildamd=true" >> $GITHUB_OUTPUT
+        else
+          echo "AMD installer already exists, skipping"
+          echo "buildamd=false" >> $GITHUB_OUTPUT
+        fi
+    - name: Checkout Version
+      if: steps.check.outputs.buildamd == 'true' || steps.check.outputs.buildarm == 'true'
+      uses: actions/checkout@v3
+      with:
+        ref: ${{steps.getversion.outputs.version}}
+    - name: Set up Go
+      if: steps.check.outputs.buildamd == 'true' || steps.check.outputs.buildarm == 'true'
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.18'
```

Comment:
Defining this hear seems it may be a maintenance headache.  In other words, maintainers are highly likely to forget to update this version when necessary/appropriate.  It will also be difficult to coordinate with the version used in CI VM's by cirrus - meaning what we test may not == what we build :disappointed:   Lastly, it may cause problems if a maintainer does a manual run of the job, specifying a `version` input that's incompatible with the hard-coded golang version :cry: 

Off-hand I don't have a good solution for this that's not complex or likely to flake due to job synchronization with Cirrus.  Maybe it's not a big deal with MacOS builds, but I can say from experience the damn golang version changes about every time with new Fedora CI VM images.  Maybe it's something that can be fixed in a follow-up PR?  Maybe a golang expert knows a clever way we could solve this?

For reference, the "simpelest" (imperfect) solution I can think of is to add steps that:
1. Run `podman version` in `quay.io/podman/upstream:latest`
2. Extract the golang version from the output
3. Use that value as the input to the `actions/setup-go` action

Note: This is imperfect because 'latest' is unlikely to coorespond to the MacOS version being built in this job.  However, maybe the gloang version used is "close enough"?

---

Slightly more complex / more accurate (still imperfect) solution:

1. Look up the `IMAGE_SUFFIX` from the checked out `.cirrus.yml` file
2. run `go env GOVERSION` in `quay.io/libpod/fedora_podman:$IMAGE_SUFFIX`
3. parse and extract the golang version from the output.
4. Use that value as the input to the `actions/setup-go` action

Reply:
Would it be enough to use the go version specified in the go.mod? That would be pretty easy to extract and set as the go version.

Reply:
Note: I see we have this same problem in `.github/workflows/upload-win-installer.yml` as well.

> the go version specified in the go.mod

I've considered this in the past.  What I found is, it's a specification not a discrete value.  It also reflects the _minimum_ version, not the one we tested with.

I also considered teaching renovate to manage the version.  That would keep it updated, but not necessarily in-sync with what we test.

This might just be one of those super-thorny problems it's better to stick your head in the sand for now.  Clearly it will need fixing in two workflows, so maybe that can be done in a future PR.

Reply:
Thinking more about this.  Let's worry about it later.  I can start a discussion on the topic to see if perhaps somebody has a simple way to solve it - or can confirm it's a non-problem.

Reply:
Opened https://github.com/containers/podman/discussions/19404